core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Team.hpp

```diff
@@ -0,0 +1,394 @@
+    ValueInit::init(a_functor, &tmp);
+    static constexpr bool UseReducer = is_reducer_v<ReducerType>;
+
+    if constexpr (!UseReducer) {
+#pragma acc parallel loop gang vector reduction(+ : tmp) copyin(a_functor)
+      for (int i = 0; i < league_size; i++) {
+        int league_id = i;
+        typename Policy::member_type team(league_id, league_size, 1,
+                                          vector_length);
+        a_functor(team, tmp);
+      }
+      m_result_ptr[0] = tmp;
+    } else {
+      OpenACCReducerWrapperTeams<ReducerType, FunctorType, Policy,
+                                 TagType>::reduce(tmp, m_policy, a_functor);
+      m_result_ptr[0] = tmp;
+    }
+  }
+
+  template <class ViewType>
+  inline ParallelReduce(
+      const FunctorType& arg_functor, const Policy& arg_policy,
+      const ViewType& arg_result_view,
+      std::enable_if_t<Kokkos::is_view<ViewType>::value, void*> = nullptr)
+      : m_functor(arg_functor),
+        m_policy(arg_policy),
+        m_reducer(InvalidType()),
+        m_result_ptr(arg_result_view.data()) {}
+
+  inline ParallelReduce(const FunctorType& arg_functor,
+                        const Policy& arg_policy, const ReducerType& reducer)
+      : m_functor(arg_functor),
+        m_policy(arg_policy),
+        m_reducer(reducer),
+        m_result_ptr(reducer.view().data()) {}
+};
+
+namespace Kokkos {
+
+// Hierarchical Parallelism -> Team thread level implementation
+#pragma acc routine seq
+template <typename iType, class Lambda, typename ValueType>
+KOKKOS_INLINE_FUNCTION std::enable_if_t<!Kokkos::is_reducer<ValueType>::value>
+parallel_reduce(const Impl::TeamThreadRangeBoundariesStruct<
+                    iType, Impl::OpenACCTeamMember>& loop_boundaries,
+                const Lambda& lambda, ValueType& result) {
+  ValueType tmp = ValueType();
+#pragma acc loop seq
+  for (iType i = loop_boundaries.start; i < loop_boundaries.end; i++)
+    lambda(i, tmp);
+  result = tmp;
+}
+
+#pragma acc routine seq
+template <typename iType, class Lambda, typename ReducerType>
+KOKKOS_INLINE_FUNCTION std::enable_if_t<Kokkos::is_reducer<ReducerType>::value>
+parallel_reduce(const Impl::TeamThreadRangeBoundariesStruct<
+                    iType, Impl::OpenACCTeamMember>& loop_boundaries,
+                const Lambda& lambda, ReducerType& result) {
+  using ValueType = typename ReducerType::value_type;
+
+  ValueType tmp = ValueType();
+#pragma acc loop seq
+  for (iType i = loop_boundaries.start; i < loop_boundaries.end; i++)
+    lambda(i, tmp);
+  result = tmp;
+}
+
+// FIXME_OPENACC: custom reduction is not implemented.
+template <typename iType, class Lambda, typename ValueType, class JoinType>
+KOKKOS_INLINE_FUNCTION void parallel_reduce(
+    const Impl::TeamThreadRangeBoundariesStruct<iType, Impl::OpenACCTeamMember>&
+        loop_boundaries,
+    const Lambda& lambda, const JoinType& join, ValueType& init_result) {
+  static_assert(Kokkos::Impl::always_false<Lambda>::value,
```

Comment:
`always_false` doesn't even exist, right?

Reply:
You are correct; `always_false` does not exist anymore; I will fix this issue.
Thanks you for catching this error.

Reply:
Have you even tested this branch locally?

Reply:
Yes, but the compiler couldn't catch this error since no template instantiation occurred, which was the purpose of this error checking code, catching unsupported template instantiation.